### ISSUE TITLE
Deploying app to Heroku fails due to Database Type

### ISSUE THREAD

**Issue by ErinRenNumber1** [Shopify/shopify-app-cli#802]
# Issue summary
I created a rails app, and tried to deploy it to Heroku, and the command fails.

Write a short description of the issue here ↓
I first created the app with the default DB (SQLite).  Upon deploy, it fails with this:
```
┏━━ Validating application... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
┃ ⠼ Checking database type...
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ (2.61s) ━━
✗ Heroku does not support deployment using the SQLite database system.
Change the database type using rails db:system:change --to=[new_db_type]. For more info:
https://gorails.com/episodes/rails-6-db-system-change-command
```

Then, I have changed the db type to postgresql: `rails db:system:change --to=postgresql`

When I tried to deploy, I get another error:
```
MacBook-Pro-99:scripto erinren$ shopify deploy heroku
┏━━ Validating application... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
┃ ⠋ Checking database type...
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ (4.3s) ━━
✗ A problem was encountered while checking your database type.
```

## Specifications

- App type: rails
- Operating System: macos
- Shell: bash
- Ruby version (ruby -v): ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-darwin19]

**Comment by paulomarg** [comment]
This happens because the rails task used to check the DB fails if it hasn't been migrated yet, so it would only happen if the app hasn't actually been run by the time the deploy is done - which is unlikely.

This isn't a bug since there is actually a problem with the database, but we could tweak the error message to indicate more clearly that the developer should make sure their DB is initialized.

**Comment by ErinRenNumber1** [comment]
Yeah, that's fair.  As discussed yesterday, CLI is targeted for experienced app developers.  Some of us don't have those kind of expertise, and these error messages are bit troubleshoot.  Thanks!

**Comment by derrickrc** [comment]
I didn't get the exact same error as you, but I was gettin ga db error at the OAuth callback step telling me the shop table wasn't found. I had followed the sqlite to pg db migration instructions from Heroku here: https://devcenter.heroku.com/articles/sqlite3

After I ran  heroku run rake db:migrate, it worked.

**Comment by paulomarg** [comment]
Yes @derrickrc, that is the right way to fix it. This error will only happen if you haven't set up your db (by running `db:migrate`) before trying to deploy the app. We will improve the error message so it reminds you to run `db:migrate` if it happens.
